activestorage/app/models/active_storage/checksum.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+class ActiveStorage::Checksum # :nodoc:
```

Comment:
Perhaps it would make sense to implement `to_s` and `to_str` to return `@digest`?

Or perhaps it's better to fail if `checksum` is used as a string?

Reply:
I only saw the one test fail in ActionText and that looked like a test expectation and not something that would have actual functionality. Is it more of an issue if we return `@digest` and it is not an MD5 digest as expected?

Reply:
That's why I'm a bit uncomfortable making such a change in Active Storage, I haven't used it enough in production myself to have a good idea of what users do with `blob.checksum`.

I suspect it's not directly access by the vast majority of users, but I don't have a strong opinion.

Reply:
I can see the backwards compatibility case for MD5 digest with to_s, but I am not sure I have a strong on to_s return values beyond that.

 I may lean toward using `dump`.  We would match the value stored in the checksum database column and would be accepted by the direct upload action param.
```
class ActiveStorage::Checksum
...
def to_s
  self.class.dump(self)
end
```

or we could raise an error if not MD5... but this seems more disruptive.
```
class ActiveStorage::Checksum
...
def to_s
  digest if algorithm == :MD5
  raise StandardError # or some other type?
end
```